### PR TITLE
Fix auto-update on macOS not working

### LINKED ISSUE #4049: Auto-update failed 5.1.3 -> 5.2.5
I launched Lens 5.1.3, got the message to update to 5.2.5, clicked "Yes". Lens quit, but did not update. Attempting to run Lens again produces the system error `The application "Lens" is not open anymore.`

`ps` shows the following process running:
```
/Applications/Lens.app/Contents/Frameworks/Squirrel.framework/Resources/ShipIt com.electron.kontena-lens.ShipIt /Users/nuru/Library/Caches/com.electron.kontena-lens.ShipIt/ShipItState.plist
```

Killing that process did not help. Attempting to re-install from DMG failed with "The operation can’t be completed because the item “Lens” is in use". Looks like I'm going to have to reboot the machine to clear the problem.

**Environment (please complete the following information):**
- Lens Version: 5.1.3 -> 5.2.5
- OS: macOS 10.15.7 (intel)
- Installation method (e.g. snap or AppImage in Linux): DMG

### LINKED ISSUE #4140: autoupdate fails on alpha channel
In trying to update from 5.3.0-alpha.2 to alpha.3 these messages are logged:

info:  :black_small_square: [WINDOW-MANAGER]: Main window has reported that it has loaded +29ms
info:  :black_small_square: Update for version 5.3.0-alpha.2.20211013.1 is not available (latest version: 5.2.5-latest.20211001.2, downgrade is disallowed). +520ms
info:  :black_small_square: [UPDATE-CHECKER]: update not available from latest, will check null channel next +0ms
error:  :black_small_square: Error: Error: Channel must be a string, but got: null
  at Object.newError (/Applications/Lens.app/Contents/Resources/app.asar/node_modules/electron-updater/node_modules/builder-util-runtime/out/index.js:47:19)
  at MacUpdater.set channel [as channel] (/Applications/Lens.app/Contents/Resources/app.asar/node_modules/electron-updater/out/AppUpdater.js:114:46)
  at MacUpdater.<anonymous> (/Applications/Lens.app/Contents/Resources/app.asar/static/build/main.js:1:264746)
  at MacUpdater.emit (events.js:315:20)
  at MacUpdater.EventEmitter.emit (domain.js:467:12)
  at MacUpdater.doCheckForUpdates (/Applications/Lens.app/Contents/Resources/app.asar/node_modules/electron-updater/out/AppUpdater.js:316:18)
  at runMicrotasks (<anonymous>)
  at processTicksAndRejections (internal/process/task_queues.js:93:5)
  at async du (/Applications/Lens.app/Contents/Resources/app.asar/static/build/main.js:1:264867)
  at async /Applications/Lens.app/Contents/Resources/app.asar/static/build/main.js:1:264775 +1ms
error:  ┏ [UPDATE-CHECKER]: failed with an error +0ms
error:  ┗ [1] { error: 'Error: Channel must be a string, but got: null' }

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

fixes #4140 

Can be reverted when `electron-updater` is released with a fix.

### CODE REVIEW

File: src/main/app-updater.ts [hunk 1]
```diff
@@ -65,7 +71,12 @@ export const startUpdateChecking = once(function (interval = 1000 * 60 * 60 * 24
 
   const userStore = UserStore.getInstance();
 
-  autoUpdater.logger = logger;
+  autoUpdater.logger = {
+    info: message => logger.info(`[AUTO-UPDATE]: electron-updater: ${message}`),
+    warn: message => logger.warn(`[AUTO-UPDATE]: electron-updater: ${message}`),
+    error: message => logger.error(`[AUTO-UPDATE]: electron-updater: ${message}`),
+    debug: message =>  logger.debug(`[AUTO-UPDATE]: electron-updater: ${message}`),
```
Comment (Nokel81) [inline src/main/app-updater.ts:78]: Not strictly necessary but makes the logs easier to read

**Review by jakolehm — APPROVED** [review]
LGTM

File: src/main/app-updater.ts [hunk 2]
```diff
@@ -41,8 +41,14 @@ function handleAutoUpdateBackChannel(event: Electron.IpcMainEvent, ...[arg]: Upd
     if (arg.now) {
       logger.info(`${AutoUpdateLogPrefix}: User chose to update now`);
       setImmediate(() => {
-        app.removeAllListeners("window-all-closed");
-        WindowManager.getInstance().destroy();
```
Comment (jim-docker) [inline src/main/app-updater.ts:45]: So no need to do this for any OS?

Comment (Nokel81) [reply src/main/app-updater.ts:45]: Nope, that was added by me in an attempt to debug this.

**Pull request merged**